src/app/user-sign-up/register/register.component.ts

```diff
@@ -26,20 +32,77 @@ export class RegisterComponent implements OnInit {
     })
   }
 
-  // getter methods
+  /**
+   * name method are used to get name
+   * @retun name
+   *  */
   get name(){
     return this.registerForm.get('name');
   }
+  /**
+   * email method are used to get email
+   * @retun email
+   *  */
   get email(){
     return this.registerForm.get('email');
   }
+  /**
+   * password method are used to get password
+   * @retun password
+   *  */
   get password(){
     return this.registerForm.get('password');
   }
-
-  // Form submit method
+  /**
+   * addUser method are used to subscribe data
+   * @param new_user : User_register
+   *  */ 
+  addUser(new_user: User_register) {
+    this.add_users.addUsers(new_user).subscribe(
+      (x) => {
+        alert("Registration Succesful");
+        this.router.navigate(['/login']);
+      },
+      (error: HttpErrorResponse) => {
+        if (error.status === 403) {
+          alert("User already exists");
+        }
+        else if (error.status === 401) {
+          alert("Unauthorized");
+        }
+        else if (error.status === 400) {
+          alert("Badrequest");
+        }
+        else if (error.status === 404) {
+          alert("Not Found");
+        }
+        else if (error.status === 500) {
+          alert("Server error");
+        }
+        else if (error.status === 501) {
+          alert("Server error");
+        }
+      }
+    );
+  }
+  // Submit method
+  /**
+   * Submit method
+   * @returns void
+   */
   onSubmit(): void{
+    const formData = this.registerForm.value;
+    const new_user : User_register ={
+     userName : formData.name,
+     emailAddress : formData.email,
+     password : formData.password,
+     appName:'Splitwise',
+   }
+   /**
+   * call add user method
```

Comment:
If you are adding comment inside the method then it should be in '//' or /*---*/ format.